src/trace.cpp

```diff
@@ -40,18 +40,8 @@ trace_t::trace_t(uint64_t trace_id_,
                  const std::string& rpc_name_):
     trace_id(trace_id_),
     state({span_id_, parent_id_, rpc_name_}),
-    previous_state(boost::none)
+    previous_state()
 {
-    auto check_range = [](uint64_t value) -> bool {
-        static const uint64_t max = (1ull << 63) - 1;
-        return value <= max;
-    };
-
-    // Check that values are in valid range.
-    if(!check_range(trace_id) || !check_range(state.span_id) || !check_range(state.parent_id)) {
-        throw cocaine::error_t("invalid trace parameters: %llu %llu %llu", trace_id, state.span_id, state.parent_id);
-    }
-
```

Comment:
Because now we can handle values in full uint64 range

______________________________
С уважением, Антон Матвеенко
antmat@nigma.ru

> 4 сент. 2015 г., в 17:18, Evgeny Safronov <notifications@github.com> написал(а):
> 
> In src/trace.cpp:
> 
> >  {
> > -    auto check_range = [](uint64_t value) -> bool {
> > -        static const uint64_t max = (1ull << 63) - 1;
> > -        return value <= max;
> > -    };
> > -
> > -    // Check that values are in valid range.
> > -    if(!check_range(trace_id) || !check_range(state.span_id) || !check_range(state.parent_id)) {
> > -        throw cocaine::error_t("invalid trace parameters: %llu %llu %llu", trace_id, state.span_id, state.parent_id);
> > -    }
> > -
> Why have you dropped sanity checks?
> 
> —
> Reply to this email directly or view it on GitHub.
>